sound/soc/sof/trace.c

```diff
@@ -135,10 +356,15 @@ static const struct file_operations sof_dfs_trace_fops = {
 static int trace_debugfs_create(struct snd_sof_dev *sdev)
 {
 	struct snd_sof_dfsentry *dfse;
+	int ret;
 
 	if (!sdev)
 		return -EINVAL;
 
+	ret = trace_debugfs_filter_create(sdev);
```

Comment:
if we return an error code from `trace_debugfs_create()`, `snd_sof_init_trace()` will fail too and leave the system with no SOF tracing. This isn't fatal for SOF probing / functionality, but maybe it's better to fail filter initialisation "silently." In fact I'd rather change `snd_sof_init_trace()` to not fail if `trace_debugfs_create()` fails, but that's something for a different PR. And anyway, if the system fails such functions, which in this case means it's really low on memory, many other things will fail too...

Reply:
@lyakh as you have spotted, possibility of failure `trace_debugfs_filter_create()` in fully functional system in quite low, and even if it will be 'silent failure', then following line is going to fail too (`dfse = devm_kzalloc(sdev->dev, sizeof(*dfse), GFP_KERNEL)`), so system will be without SOF tracing at all. In my opinion it's not an argument to drop return code check and relay on other things fails. I can replace `return ret` with `dev_err`, but because of low fail probability, it shouldn't change much for end-user.